js/components/portal-dashboard/all-responses-popup/popup-class-nav.tsx

```diff
@@ -0,0 +1,137 @@
+    const { showDialog } = this.state;
+
+    return (
+      <React.Fragment>
+        <div className={`${css.popupClassNav}`}>
+          {this.renderViewListOptions()}
+          < div className={`${cssClassNav.classNav} ${css.popupClassNavControllers}`} data-cy="class-nav" >
+            <AnonymizeStudents setAnonymous={setAnonymous} />
+            <NumberOfStudentsContainer studentCount={studentCount} />
+            {this.renderStudentFilter()}
+            {this.renderSpotlightToggle()}
+          </div >
+        </div>
+        {showDialog && < SpotlightDialog handleCloseDialog={this.closeShowDialog} />}
+      </React.Fragment>
+    );
+  }
+
+  private renderStudentFilter = () => {
+    const items: SelectItem[] = [{ action: SORT_BY_NAME, name: "All Students" }];
+    const { setStudentFilter, trackEvent } = this.props;
+    return (
+      <div className={cssClassNav.studentSort}>
+        <CustomSelect
+          items={items}
+          onSelectItem={setStudentFilter}
+          trackEvent={trackEvent}
+          iconId={"icon-sort"}
+          dataCy={"sort-students"}
+        />
+      </div>
+    );
+  }
+  private renderViewListOptions() {
+    let listByStudentClasses: string = css.toggle + " " + css.listByStudents,
+      listByQuestionsClasses: string = css.toggle + " " + css.listByQuestions;
+
+    if (!this.state.inQuestionMode) {
+      listByStudentClasses += "  " + css.selected;
+    } else {
+      listByQuestionsClasses += " " + css.selected;
+    }
+
+    return (
+      <div className={`${css.viewListOption} ${css.columnHeader}`}>View list by:
+        <div className={listByStudentClasses} data-cy="list-by-student-toggle" onClick={this.handleStudentsToggleClick}>
+          <svg className={`${css.optionIcon}  `}>
+            <use xlinkHref={"#icon-student-view"} />
+          </svg>
+        </div>
+        <div className={listByQuestionsClasses} data-cy="list-by-questions-toggle" onClick={this.handleQuestionsToggleClick}>
+          <svg className={`${css.optionIcon}`}>
+            <use xlinkHref={"#icon-question-view"} />
+          </svg>
+        </div>
+      </div>
+    );
+  }
+
+  private renderSpotlightToggle() {
+    let spotLightContainerClasses: string = css.spotlightContainer;
+    if (this.state.isSpotlightOn && this.state.studentSelected) { spotLightContainerClasses += " " + css.spotlightOn; }
+    return (
+      <div className={`${css.spotlightToggle}`} onClick={this.handleSpotlightClick} data-cy="spotlight-toggle">
+        <div className={spotLightContainerClasses}>
+          <svg className={`${css.spotlightIcon}`}>
+            <use xlinkHref={"#icon-spotlight"} />
+          </svg>
+        </div>
+        <span>Spotlight selected</span>
+      </div>
+    );
+  }
+
+  private handleStudentsToggleClick = () => {
+    this.setState({
+      inQuestionMode: false
+    });
+  }
+  private handleQuestionsToggleClick = () => {
+    this.setState({
+      inQuestionMode: true
+    });
+  }
```

Comment:
handleStudentsToggleClick and handleQuestionsToggleClick could be refactored into a single function that takes in a boolean, but isn't necessary.

Reply:
Something like this:
```
  private setQuestionMode = (value: boolean) => () => {
    this.setState({
      inQuestionMode: value
    });
  }
```

And then, you can simply add JSX:
```
<div className={listByStudentClasses} data-cy="list-by-student-toggle" onClick={this.setQuestionMode(false)}>

<div className={listByQuestionsClasses} data-cy="list-by-questions-toggle" onClick={this.setQuestionMode(true)}>
```